airqo-platform/AirQo-api
# [DP-171] work on predict microservice

# work on predict microservice

**_WHAT DOES THIS PR DO?_**
- Change directory structure of predict microservice (move jobs to one directory)
- Refactor predict training job
- Add mlops configuration using mlflow
- Set automatic model registering, experiment tracking, logging parameters for each model run
- Automatic model backup and update
- Closes #804 

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
- [DP-171]
- [PLAT-1072](https://airqoteam.atlassian.net/browse/PLAT-1072)

**_WHAT IS THE LINK TO THE PR BRANCH_**
- https://github.com/airqo-platform/AirQo-api/tree/predict-train-refactor

**_HOW DO I TEST OUT THIS PR?_**

1. check out this branch and `cd` to `src/predict/jobs/train/`
2. using python virtual environment, run `pip install -r requirements.txt`. Make sure you have the `.env` file.
3. Run `python main.py` and wait for the process to finish. *NB: set the environment to `staging` to avoid overwriting models in production*.
4. Check the result on mlflow server: `http://23.251.144.212:5000`. You can also check the `model.pkl` and `all_channels.pkl` uploaded in google storage bucket: `airqo_prediction_bucket_stage`.

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- NONE

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- NONE

**_ARE THERE ANY RELATED PRs?_**
- NONE



[DP-171]: https://airqoteam.atlassian.net/browse/DP-171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
